examples/tutorials/vectors.py

```diff
@@ -0,0 +1,459 @@
+# projection of the continental US. The x values represent the
+# longitude and y values represent the latitude where the vector starts.
+
+# create a plot with coast, Mercator projection (M) over the continental US
+fig = pygmt.Figure()
+fig.coast(
+    region=[-127, -64, 24, 53],
+    projection="M15c",
+    frame="ag",
+    borders=1,
+    area_thresh=4000,
+    shorelines="0.25p,black",
+    land="grey",
+    water="lightblue",
+)
+
+style = "v0.6c+bc+ea+a30"
+fig.plot(
+    x=-110,
+    y=40,
+    style=style,
+    pen="1p",
+    color="red3",
+    direction=[[-25], [3]],
+)
+
+# vector specifications structured as: [x_start, y_start, direction_degrees, length]
+vector_2 = [-82, 40.5, 138, 3]
+vector_3 = [-71.2, 45, -115.7, 6]
+# Create a list of lists that include each vector information
+vectors = np.array([vector_2] + [vector_3])
+
+fig.plot(
+    data=vectors,
+    style=style,
+    pen="1p",
+    color="yellow",
+)
+
+fig.show()
+
+########################################################################################
+# Another example of plotting cartesian vectors over a coast plot. This time
+# a Transverse Mercator projection is used. Additionally, ``numpy.linespace``
+# is used to create 5 vectors with equal stops.
+
+# create a plot with coast, Mercator projection (M) over the continental US
+fig = pygmt.Figure()
+fig.coast(
+    region=[20, 50, 30, 45],
+    projection="T35/12c",
+    frame=True,
+    borders=1,
+    area_thresh=4000,
+    shorelines="0.25p,black",
+    land="lightbrown",
+    water="lightblue",
+)
+
+x = np.linspace(36, 42, 5)  # x values = [36.  37.5 39.  40.5 42. ]
+y = np.linspace(39, 39, 5)  # y values = [39. 39. 39. 39.]
+direction = np.linspace(-90, -90, 5)  # direction values = [-90. -90. -90. -90.]
+length = np.linspace(1.5, 1.5, 5)  # length values = [1.5 1.5 1.5 1.5]
+
+fig.plot(
+    x=x,
+    y=y,
+    style="v0.4c+ea",
+    pen="0.6p",
+    color="red3",
+    direction=[direction, length],
+)
+
+
+fig.show()
+
+########################################################################################
+# Plot Circular Vectors
+# ---------------------
+#
+# When plotting circular vectors, there are 5 values that should be included in
+# the list that is passed through ``np.array()`` in order to create a valid plot.
+# The first two values in ``circular_vector_1`` represent the origin of the
+# circle that will be plotted. The next value is the radius which is represented
+# on the plot in centimeters. Finally, the last two values represent the degree
+# at which the plot will start and stop. In this example, the result shown is the
+# left half of a circle as the plot starts at 90 degrees and goes until 270.
+# It is important to note that when plotting circular vectors, the style value
+# should begin with an ``m``.
+
+# vector specifications structured as: [x_start, y_start, radius, degree_start, degree_stop]
+circular_vector_1 = [0, 0, 5, 90, 270]
+
+data = np.array([circular_vector_1])
```

Comment:
This paragraph gives me the impression that users have to pass a numpy array to plot circular vectors. It's not true. A list of 5 values are also accepted.

Reply:
Thanks for the feedback! I can definitely see what you mean about the paragraph and we'll work on clarifying that. I'm also curious about the list of 5 values. I tried this code and I was getting `AttributeError: 'list' object has no attribute 'flags'`

```circular_vector_1 = [0, 0, 5, 90, 270]

fig = pygmt.Figure()
fig.plot(
    region=[-10, 10, -10, 10],
    projection="X20c",
    frame="ag",
    data=circular_vector_1,
    style="m0.5c+ea",
    pen="2p",
    color="red3",
)
fig.show()
```
Would I use a different parameter than `data`?

Reply:
Hmm, maybe a bug.

Reply:
It's really weird, we initially thought that the user has to use a NumPy array by design, which is why we did the same thing in the cartesian examples. What do you suggest that we do here?

Also, I made the other changes regarding the image sizes, please let us know if there are still issues with the current projections. Thanks!

Reply:
This one should work. The list/array passed to GMT must be 2d. In your code, `circular_vector_1` is a 1d array and is passed to GMT.
```
fig = pygmt.Figure()
fig.plot(
    region=[-10, 10, -10, 10],
    projection="X10c",
    frame="ag",
    data=[circular_vector_1],
    style="m0.5c+ea",
    pen="2p",
    color="red3",
)
fig.show()
```

Reply:
I just opened an issue for it: https://github.com/GenericMappingTools/pygmt/issues/1132

Reply:
Thank you! Depending on our plans in the Software Innovations course, we might continue to work with this project and the issue you just opened. In the meantime, are there any other requested changes for this tutorial?

Reply:
> In the meantime, are there any other requested changes for this tutorial?

Thanks a lot for this amazing tutorial. I think it is a very important one and I have leant a lot by reviewing it. Since it is a little long, I may still need a few days to review the whole tutorial.

Reply:
> > In the meantime, are there any other requested changes for this tutorial?
> 
> Thanks a lot for this amazing tutorial. I think it is a very important one and I have leant a lot by reviewing it. Since it is a little long, I may still need a few days to review the whole tutorial.

I'm glad that you find this tutorial helpful. We'll be working on adding your requested changes as they come up.